### PR TITLE
update to Android Studio 3.1.2

### PR DESCRIPTION
**Pull request by hannesa2**
This PR updates to most recent stable Android Studio 3.1.2

### CODE REVIEW

File: MPChartExample/src/com/xxmassdeveloper/mpchartexample/notimportant/DemoBase.java [hunk 1]
```diff
@@ -36,6 +37,8 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         mTfRegular = Typeface.createFromAsset(getAssets(), "OpenSans-Regular.ttf");
         mTfLight = Typeface.createFromAsset(getAssets(), "OpenSans-Light.ttf");
+
+        Log.d("OpenActivity", this.getClass().getName());
```
Comment (MGaetan89) [inline MPChartExample/src/com/xxmassdeveloper/mpchartexample/notimportant/DemoBase.java]: I guess there is no reason for this change either (and its import).

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Mon Nov 20 11:59:54 CET 2017
+#Wed Apr 25 08:04:33 CEST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
Comment (MGaetan89) [inline gradle/wrapper/gradle-wrapper.properties]: The latest version of Gradle is `4.7`.

Comment (hannesa2) [reply gradle/wrapper/gradle-wrapper.properties]: gradle build tools 3.1.2 doesn't work with gradle > 4.5

**Comment by hannesa2** [comment]
Sorry there were to much local changes in the PR. I changed this.
Btw, a lot of other things could be changed, but this is out of scope here
eg:
image

**Comment by almic** [comment]
It does seem that new gradle versions are available, however since Android Studio is only automatically updating to 4.4, that should be fine for now. I just built this version and it worked, merging.

**Comment by hannesa2** [comment]
gradle build tools 3.1.2 doesn't work with gradle > 4.5 

> Failed to apply plugin [id 'com.android.application']
   > Configuration on demand is not supported by the current version of the Android Gradle plugin since you are using Gradle version 4.6 or above. Suggestion: disable configuration on demand by setting org.gradle.configureondemand=false in your gradle.properties file or use a Gradle version less than 4.6.

First suggestion doesn't work too. 
If you have a better way, let us know

**Pull request merged**